src/main/java/org/elasticsearch/script/expression/FieldDataValueSource.java

```diff
@@ -49,13 +48,24 @@ public FunctionValues getValues(Map context, LeafReaderContext leaf) throws IOEx
     }
 
     @Override
-    public boolean equals(Object other) {
-        return fieldData.equals(other);
+    public boolean equals(Object o) {
```

Comment:
I'm going to leave this as I'd prefer to not modify the generated methods unless it's necessary for code correctness.  This way as the classes change we can simply continue to regenerate the methods and still have them be consistent.